### PR TITLE
PUB-544 - Manual Upload Summary 

### PR DESCRIPTION
**Pull request by commitBlob**
### JIRA link ###
https://tools.hmcts.net/jira/browse/PUB-544

### Change description ###
- add manual upload summary screen
- add upload success screen
- cookies 🍪 are used to pass data back to manual upload page
- superagent is used for file upload as axios struggled to upload multipart data
- remove file and home links do not lead anywhere at the moment as dependant screens have not been developed yet

**NOTE:**
This ticket is reliant on PUB-546, hence it should be only played after previous one is signed off.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```
summary
witherror
success

### CODE REVIEW

File: src/main/controllers/ManualUploadController.ts [hunk 1]
```diff
@@ -41,6 +41,9 @@ export default class ManualUploadController {
       req.body['court'] = await manualUploadService.appendCourtId(req.body['input-autocomplete']);
       req.body['artefactType'] = 'LIST'; //Agreed on defaulting to only option available until more types become ready
       req.body['fileName'] = req.file['originalname'];
+      req.body['display-from'] = manualUploadService.joinDateValues(req.body, 'display-date-from');
```
Comment (kurt-uk) [inline src/main/controllers/ManualUploadController.ts:44]: theres already this method in the manual upload servivce called `buildDate`

File: src/main/service/manualUploadService.ts [hunk 2]
```diff
@@ -102,7 +105,7 @@ export class ManualUploadService {
     return 'Please enter a valid date';
   }
 
-  private validateDateRange(dateFrom: string, dateTo: string) {
+  private validateDateRange(dateFrom: string, dateTo: string): string | null {
```
Comment (kurt-uk) [inline src/main/service/manualUploadService.ts:108]: ⭐

File: src/main/service/manualUploadService.ts [hunk 3]
```diff
@@ -115,4 +118,83 @@ export class ManualUploadService {
...
+      } else {
+        return fs.readFileSync(`./manualUpload/tmp/${fileName}`);
+      }
+    } catch (err) {
+      console.error(`Error while reading the file ${err}.`);
+      return null;
+    }
+  }
+
+  public joinDateValues(formData: any, dateName: string): string {
```
Comment (kurt-uk) [inline src/main/service/manualUploadService.ts:159]: this method already exists called `buildDate` line 84~

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/controllers/ManualUploadController.ts (hunk 1 above)
Comment (commitBlob) [reply src/main/controllers/ManualUploadController.ts:44]: didn't see that, removing mine

File: src/main/service/manualUploadService.ts (hunk 3 above)
Comment (commitBlob) [reply src/main/service/manualUploadService.ts:159]: true, didn't see that, removed

File: src/main/service/manualUploadService.ts [hunk 4]
```diff
@@ -1,32 +1,39 @@
...
-      {text: 'Crown Daily List', value: 'CROWN_DAILY_LIST'},
-      {text: 'Crown Firm List', value: 'CROWN_FIRM_LIST'},
-      {text: 'Crown Warned List', value: 'CROWN_WARNED_LIST'},
-      {text: 'Magistrates Public List', value: 'MAGS_PUBLIC_LIST'},
-      {text: 'Magistrates Standard List', value: 'MAGS_STANDARD_LIST'},
-    ];
+    return listSubTypes;
+  }
+
+  public getListItemName(itemValue: string): string {
```
Comment (kurt-uk) [inline src/main/service/manualUploadService.ts:35]: 🌟

**Review by kurt-uk — APPROVED** [review]

**Pull request merged**